### PR TITLE
Create ``_ExtendArgument``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Currently uncovered but necessary for `pyreverse` since it has an `append` option, which in `argparse` is the same as an `extend` option.

Added in a separate PR to keep the final `pyreverse` PR light-weight.

### CODE REVIEW

File: pylint/config/callback_actions.py [hunk 1]
```diff
@@ -52,6 +52,27 @@ def __call__(
         return None
 
 
+class _ExtendAction(argparse._AppendAction):
+    """Action that adds the value to a pre-existing list.
+
+    It is directly copied from the stdlib implementation which is only available
+    on 3.8+.
```
Comment (Pierre-Sassoulas) [inline pylint/config/callback_actions.py:59]: Maybe we could start a TODO list for when we drop 3.7 like #4683 to not forget to cleanup that kind of thing.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM !

File: pylint/config/callback_actions.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/config/callback_actions.py:59]: I think whenever we drop support for a version we're likely to do a `grep` for `3.8` `3, 8` and other combinations. But feel free to open one!

Comment (Pierre-Sassoulas) [reply pylint/config/callback_actions.py:59]: https://github.com/PyCQA/pylint/issues/6306

**Pull request merged**